File: 2958-length-of-longest-subarray-with-at-most-k-frequency/2958-length-of-longest-subarray-with-at-most-k-frequency.py
class Solution:
    def maxSubarrayLength(self, nums: List[int], k: int) -> int:
        n=len(nums)
        i=0
        j=0
        res=0
        map={}
        while j<n:
            map[nums[j]]=map.get(nums[j],0)+1
            while i<j and map[nums[j]]>k:
                map[nums[i]]-=1
                i+=1
            res=max(res,j-i+1)
            j+=1
        return res
        

File: 2958-length-of-longest-subarray-with-at-most-k-frequency/2958-length-of-longest-subarray-with-at-most-k-frequency.cpp
class Solution {
public:
    int maxSubarrayLength(vector<int>& nums, int k) {
        int n=nums.size();
        int i=0,j=0;
        unordered_map<int,int> map;
        int res=0;
        while(j<n){
            map[nums[j]]++;
            while(i<j && map[nums[j]]>k){
                map[nums[i]]--;
                i++;
            }
            res=max(res,j-i+1);
            j++;
        }
        return res;
    }
};